ReSwiftTests/StoreMiddlewareTests.swift

```diff
@@ -9,7 +9,7 @@
 import XCTest
 @testable import ReSwift
 
-let firstMiddleware: Middleware<StateType> = { dispatch, getState in
+let firstMiddleware: Middleware<Any> = { dispatch, getState in
```

Comment:
This is the part I don't understand. Would `getState` here know the type of the store's state? If so, does Middleware have to be generic over something at all?

Reply:
This middleware test doesn't use `getState` -- in middleware that needs to access state, this would be `Middleware<AppState>` or some subset protocol, IE: `struct AppState: HasName`, and then `Middleware<HasName>`.

Reply:
Alright, makes sense!